src/lib/cache.js

```diff
@@ -1,24 +1,67 @@
 import { isNull, isArray } from "lodash"
-import { Client } from "memjs"
 import config from "config"
-import { error } from "./loggers"
+import { error, verbose } from "./loggers"
+import redis from "redis"
+import url from "url"
 
-const { NODE_ENV } = process.env
+const { NODE_ENV, OPENREDIS_URL } = process.env
 const { CACHE_LIFETIME_IN_SECONDS } = config
 
-const createClient = () => {
-  if (NODE_ENV === "test") {
-    const store = {}
-    return {
-      store,
-      get: (key, cb) => cb(null, store[key]),
-      set: (key, data) => (store[key] = data),
-    }
+const isTest = NODE_ENV === "test"
+
+const VerboseEvents = ["connect", "ready", "reconnecting", "end", "warning"]
+
+function createMockClient() {
+  const store = {}
+  return {
+    store,
+    get: (key, cb) => cb(null, store[key]),
+    set: (key, data) => (store[key] = data),
+  }
+}
+
+function createRedisClient() {
+  const redisURL = url.parse(OPENREDIS_URL)
+  const client = redis.createClient({
+    host: redisURL.hostname,
+    port: redisURL.port,
+    retryStrategy: options => {
+      if (options.error) {
+        // Errors that lead to the connection being dropped are not emitted to
+        // the error event handler, so send it there ourselves so we can handle
+        // it in one place.
+        // See https://github.com/NodeRedis/node_redis/issues/1202#issuecomment-363116620
+        client.emit("error", error)
+        // End reconnecting on a specific error and flush all commands with a
+        // individual error.
+        if (options.error.code === "ECONNREFUSED") {
+          return new Error("The server refused the connection")
+        }
+      }
+      // End reconnecting after a specific timeout and flush all commands with a
+      // individual error.
+      if (options.total_retry_time > 1000 * 60 * 60) {
+        return new Error("Retry time exhausted")
+      }
+      // End reconnecting with built in error.
+      if (options.attempt > 10) {
+        return undefined
+      }
+      // Reconnect after:
+      return Math.min(options.attempt * 100, 3000)
```

Comment:
this `min` seems redundant since we break at `> 10` attempts, so the first arg will never reach over 1000. but it does allow us to vary the max attempt count without worrying i guess.